repository_service_tuf_api/metadata.py

```diff
@@ -227,7 +227,7 @@ def get_metadata_sign() -> MetadataSignGetResponse:
     # Adds support only when bootstrap is signing state
     if bs_state.bootstrap is False and bs_state.state != "signing":
         raise HTTPException(
-            status.HTTP_200_OK,
+            status.HTTP_404_NOT_FOUND,
```

Comment:
For this `get metadata sign`, I don't know if we should return 404.
We should keep 200 for this get as the resource is listed in the body.
WDYT? (cc @simi)

Reply:
> For this get metadata sign, I don't know if we should return 404.

IMO this is the same as the rest. That is, in pre-bootstrap state the resource simply does not exist. So you should return 404, also for consistency. Post-bootstrap , OTOH, if the resource does exist but is empty, it makes sense to return 200 and `{"metadata": {}}`.

For me the important thing is that I don't have to check the [status AND message](https://github.com/pypi/warehouse/blob/6dfb07dd44716fea35cb4ee8e23ee5011e0a2d73/warehouse/tuf/__init__.py#L47-L53) to know if there is an issue.



> We should keep 200 for this get as the resource is listed in the body.

I don't understand.